### PR TITLE
Stateful Client

### PR DESCRIPTION
**Pull request by jen20**
This is a continuation of the discussion on #36 now that commits have been brought across into the `jen20-stateful-client` branch.

### CODE REVIEW

File: azure/hostedservice/client.go [hunk 1]
```diff
@@ -18,24 +18,34 @@ const (
...
-	paramNotSpecifiedError = "Parameter %s is not specified."
+	errParamNotSpecified = "Parameter %s is not specified."
+	errInvalidDnsLength  = "The DNS name must be between 3 and 25 characters."
 )
 
-func CreateHostedService(dnsName, location string, reverseDnsFqdn string) (string, error) {
+//HostedServiceClient is used to manage operations on Azure Hosted Services
+type HostedServiceClient struct {
+	client *azure.Client
+}
```
Comment (hopetobelievein) [inline azure/hostedservice/client.go]: What about moving structs to one place?
I agree that we should keep <code>NewClient()</code> here

Comment (jen20) [inline azure/hostedservice/client.go]: Landed in d1c1382

File: azure/http.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	responseContent := getResponseBody(response)
+	return responseContent, nil
+}
+
+//sendAzurePostRequest sends a request to the management API using the HTTP POST method
+//and returns the response body or an error.
```
Comment (hopetobelievein) [inline azure/http.go]: It is a small thing, but method returns requestId or an error. Same in SendAzurePutRequest and SendAzureDeleteRequest

Comment (jen20) [inline azure/http.go]: @hopetobelievein not sure I follow?

Comment (hopetobelievein) [inline azure/http.go]: <code>SendAzurePostRequest<code> returns Azure request ID or an error, but in comment describing this function written "and returns the response body or an error"

**Comment by ahmetalpbalkan** [comment]
great, thanks for reopening a PR. I'll review this. :clap:

File: azure/http.go (hunk 2 above)
Comment (jen20) [inline azure/http.go]: Ahhh good catch. Landed in bd22228.

**Comment by jen20** [comment]
@ahmetalpbalkan Want me to squash this? Would likely be cleaner for the history.

**Comment by jen20** [comment]
No idea why this test is failing, it's not code I've touched. Perhaps the test data has become invalid?

**Comment by jen20** [comment]
Closing this PR in favour of the `jen20-stateful-client-squashed` branch.

**Comment by ahmetalpbalkan** [comment]
@jen20 feel free to have a copy of history in your remote but we can probably keep a PR containing all changes active here if you don't mind. Therefore squash and pushing to this branch (from your local/stateful-client to msopentech/jen20-stateful-client) would be nice. I don't suggest pushing any code directly to MSOpenTech repo directly without going through a PR. (meaning, this PR probably should have been sent from your repo to MSOpenTech repo.)

**Pull request closed without merging**